### ISSUE TITLE
Implement color-scheme handling on basis of new `.sublime-color-scheme` format

### ISSUE THREAD

**Issue by pykong** [SublimeLinter/SublimeLinter3#639]
With the dev 3149 a new color scheme format `.sublime-color-scheme` has been introduced.
The most interesting feature is the capacity of automatically overwriting the default color scheme, as it happens with other **ST** formats.

https://forum.sublimetext.com/t/dev-build-3149/32549

This immediately implies a better way of handling color schemes in **SL**, compared to the current way of modifying the current scheme. Once this new color scheme format founds its way into the stable branch of **ST**, **SL** should switch over. 
Possible **SL** could support both old and new color scheme handling in parallel and switch depending on the current sublime version.

If this is desired by the maintainers of **SL**, I can well do the coding work required.

**Comment by braver** [comment]
I think we will have to support the old style as well, unless ST does an on the fly conversion of old color schemes. However, the new approach is an excellent fit for our use case and we should try to support it ASAP and not wait for it to land in stable. That way we have a chance of testing it with early adopters and provide jps and wbond with feedback to make it the best it can be.

I’m not sure I read it correctly, but I think the new builds will also give us `—redish` and `—yellowish` variables to work with.

If you’re up for coding something, I’ll definitely help with testing and review.

**Comment by pyking2** [comment]
@braver Thanks for the fast reply.

I am up for coding it. The other new feature I am currently implementing for **SL** and you were __so__ curious about deals with the same parts of the code. As I have already cleaned up the current color scheme generation, it is the most logical thing I code support for the new color theme system as well.

*I get to coding it starting tomorrow.**
I am thankful for your offer of testing and review and most likely come back to it!

I noticed the on the fly generated scopes for easier coloring from **3148**. I have still to wrap my mind around how this would profit **SL**.

Citing jps from the mentioned forum post:
> (.sublime-color-scheme files can override rules in both other .sublime-color-scheme files, and .tmTheme files)



For me this reads that the new color scheme system can entirely replace the old one.
So as long it is implemented at all (version 3149+), we do not need to care about the actual theme file type (`.tmTheme` vs `.sublime-color-scheme`)

**Comment by braver** [comment]
> For me this reads that the new color scheme system can entirely replace the old one.

Oh wow, that’s pretty awesome. Looking forward to testing your contributions!

**Comment by thatryan** [comment]
Didn't want to open a new issue if this is my problem, currently running ST build `3154` and default linter setup I am not getting any color output. Just tiny squigglies and no color gutter dots. Is this expected? 

<img width="698" alt="cloudapp annotation 2017-11-24 at 3 36 13 pm png" src="https://user-images.githubusercontent.com/26583/33225483-54386210-d12d-11e7-8465-9bb03039e628.png">


Theme is Adaptive, and color scheme is Default/Mariana.

Thank you.

**Comment by braver** [comment]
Yes, I don’t think SL works with the new color scheme files right now, but it will in #666
